research/fedora-packages-onboarding/get-maintainers.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import json
+import sys
+from collections import defaultdict, OrderedDict
+from pathlib import Path
+
+from bodhi.client.bindings import BodhiClient
+
+from ogr import PagureService
+from ogr.services.pagure import PagureProject
+from specfile import Specfile
+
+
+bodhi_client = BodhiClient()
+RPM_SPECS = Path("./rpm-specs")
+checks = []
+
+
+def check_decorator(check_func):
+    checks.append(check_func)
+
+    def wrapper(package_name: str, specfile: Specfile, *args, **kwargs):
+        return check_func(package_name, specfile, *args, **kwargs)
+
+    return wrapper
+
+
+@check_decorator
+def check_number_of_patches(package_name: str, specfile: Specfile):
+    with specfile.patches() as patches:
+        patches_number = len(patches)
+        print(f"Number of patches: {patches_number}")
+        return patches_number <= 1
+
+
+@check_decorator
+def check_number_of_sources(package_name: str, specfile: Specfile):
+    with specfile.sources() as sources:
+        sources_number = len(sources)
+        print(f"Number of sources: {sources_number}")
+        return sources_number == 1
+
+
+@check_decorator
+def check_specfile_length(package_name: str, specfile: Specfile):
+    with specfile.sections() as sections:
+        try:
+            changelog_index = sections.find("changelog")
+        except ValueError:
+            changelog_index = None
+        line_count = sum(len(s) for s in sections[:changelog_index])
+
+    print(f"Number of lines before %changelog: {line_count}")
+    return line_count < 200
```

Comment:
magic number :)

thinking out loud: I wonder if `%build` and `%prep` length are actually the problem